File: ScrollViewExamples/AvoidHidingFromKeyboard/AvoidHidingFromKeyboard/ViewController.m
//
//  ViewController.m
//  AvoidHidingFromKeyboard
//
//  Created by Erik Engheim on 18/09/14.
//  Copyright (c) 2014 Translusion. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property (strong, nonatomic) IBOutlet UIScrollView *scrollView;
@property (strong, nonatomic) IBOutlet UISwitch *switchButton;

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
	// Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
- (IBAction)scrollMe:(id)sender {
//    [_scrollView scrollRectToVisible:self.switchButton.frame animated:YES];
    [_scrollView setContentOffset:CGPointMake(0, 50) animated:YES];

}

- (IBAction)scrollItBack:(id)sender {
    [_scrollView setContentOffset:CGPointMake(0, 0) animated:YES];
}

@end

File: ScrollViewExamples/Zoomer/Zoomer/ViewController.swift
//
//  ViewController.swift
//  Zoomer
//
//  Created by Erik Engheim on 22/05/15.
//  Copyright (c) 2015 Pexip. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var scrollView: UIScrollView!
    @IBOutlet weak var zoomInFactor: UITextField!
    @IBOutlet weak var zoomOutFactor: UITextField!
    @IBOutlet weak var xOffsetField: UITextField!
    @IBOutlet weak var yOffsetField: UITextField!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


    @IBAction func zoomIn(sender: AnyObject) {
        
    }
    
    @IBAction func zoomOut(sender: AnyObject) {
        
    }
    
    @IBAction func scrollToOffset(sender: AnyObject) {
        var y = yOffsetField.text.toInt() ?? 0
        self.scrollView.setContentOffset(CGPoint(x: xOffsetField.text.toInt() ?? 0, y: y), animated: true)
    }
    
    @IBAction func sliderChanged(sender: UISlider) {
        var screenHeight: Float = Float(UIScreen.mainScreen().bounds.height)
        var offset: Int = Int(sender.value * screenHeight)
        self.yOffsetField.text = "\(offset)"
    }
}

